### PR TITLE
Fix 1735

### LINKED ISSUE #1735: Cannot query cypher fields on root connections when sort is not provided as an argument
If you query a root level connection where the node has a cypher field, you'll get an error.

```json
{
  "errors": [
    {
      "message": "Cannot read properties of undefined (reading 'find')",
      "path": [
        "organizationsConnection"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "message": "Cannot read properties of undefined (reading 'find')",
          "path": [
            "organizationsConnection"
          ],
          "stacktrace": [
            "TypeError: Cannot read properties of undefined (reading 'find')",
            "    at reducer (/Users/litewarp/Projects/tltf/node_modules/.pnpm/@neo4j+graphql@3.3.0_iwp6h2plez23ko5pkvojd3kjfq/node_modules/@neo4j/graphql/dist/translate/create-projection-and-params.js:233:45)",
            "    at Array.reduce (<anonymous>)",
            "    at createProjectionAndParams (/Users/litewarp/Projects/tltf/node_modules/.pnpm/@neo4j+graphql@3.3.0_iwp6h2plez23ko5pkvojd3kjfq/node_modules/@neo4j/graphql/dist/translate/create-projection-and-params.js:504:70)",
            "    at translateRead (/Users/litewarp/Projects/tltf/node_modules/.pnpm/@neo4j+graphql@3.3.0_iwp6h2plez23ko5pkvojd3kjfq/node_modules/@neo4j/graphql/dist/translate/translate-read.js:59:67)",
            "    at resolve (/Users/litewarp/Projects/tltf/node_modules/.pnpm/@neo4j+graphql@3.3.0_iwp6h2plez23ko5pkvojd3kjfq/node_modules/@neo4j/graphql/dist/schema/resolvers/query/root-connection.js:38:64)",
            "    at Object.organizationsConnection (/Users/litewarp/Projects/tltf/node_modules/.pnpm/@neo4j+graphql@3.3.0_iwp6h2plez23ko5pkvojd3kjfq/node_modules/@neo4j/graphql/dist/schema/resolvers/wrapper.js:65:12)",
            "    at executeField (/Users/litewarp/Projects/tltf/node_modules/.pnpm/graphql@16.5.0/node_modules/graphql/execution/execute.js:481:20)",
            "    at executeFields (/Users/litewarp/Projects/tltf/node_modules/.pnpm/graphql@16.5.0/node_modules/graphql/execution/execute.js:413:20)",
            "    at executeOperation (/Users/litewarp/Projects/tltf/node_modules/.pnpm/graphql@16.5.0/node_modules/graphql/execution/execute.js:344:14)",
            "    at execute (/Users/litewarp/Projects/tltf/node_modules/.pnpm/graphql@16.5.0/node_modules/graphql/execution/execute.js:136:20)"
          ]
        }
      }
    }
  ],
  "data": null
}
```
The issue is that this line expects sortInput to be an array and not null or undefined: https://github.com/neo4j/graphql/blob/e1fda9fb14f533652a3e7437733352f114daa2ac/packages/graphql/src/translate/create-projection-and-params.ts#L333

The solution is to check for existence of sortInput in the if statement, or to set sortInput to default to an empty array.

Will submit a PR later to fix.

### PR DESCRIPTION
**Pull request by litewarp**
# Description

Fixes #1735 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [X] TCK tests have been updated
- [X] Integration tests have been updated
- [ ] Example applications have been updated
- [X] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: packages/graphql/tests/integration/issues/1735.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { graphql } from "graphql";
+import { Driver } from "neo4j-driver";
```
Comment (tbwiss) [inline packages/graphql/tests/integration/issues/1735.int.test.ts:21]: We changed recently to differentiate whether the import is a type or a value (https://github.com/neo4j/graphql/pull/1698).
```suggestion
import type { Driver } from "neo4j-driver";
```

File: packages/graphql/tests/integration/issues/1735.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    type ${testActor.name} {
+        dbId: ID! @id(global: true) @alias(property: "id")
+        name: String!
+        movies: [${testMovie.name}!]! @relationship(type: "ACTED_IN", direction: OUT, properties: "MovieActorEdgeProperties")
+    }
+  `;
+    beforeAll(async () => {
+        driver = await neo4j();
```
Comment (tbwiss) [inline packages/graphql/tests/integration/issues/1735.int.test.ts:56]: We also tweaked the integration test "setup" a little bit recently. See test file [988.int.test.ts](https://github.com/neo4j/graphql/blob/dev/packages/graphql/tests/integration/issues/988.int.test.ts) for the changes that you need to make in your test file.

File: packages/graphql/tests/integration/issues/1735.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        neoSchema = new Neo4jGraphQL({ typeDefs, driver });
+        await neoSchema.checkNeo4jCompat();
+    });
+    afterAll(async () => {
+        await driver.close();
+    });
+    test("root connection queries on nodes with Cypher fields should execute without a sort argument", async () => {
+        const session = driver.session();
```
Comment (tbwiss) [inline packages/graphql/tests/integration/issues/1735.int.test.ts:64]: question: What is the purpose of this session?
if it's not needed please also clean up the try/finally block at the bottom of this file.

**Review by tbwiss — COMMENTED** [review]
Thank you @litewarp for reporting and fixing this bug!
We made some changes recently that are also reflected in the integration test, I left some comments and pointers on that.

File: packages/graphql/tests/integration/issues/1735.int.test.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                        edge: { isLead: false },
+                        node: { name: "River Phoenix" },
+                    },
+                },
+            },
+        ];
+        await graphql({
+            source,
+            schema,
+            contextValue: { driver },
```
Comment (tbwiss) [inline packages/graphql/tests/integration/issues/1735.int.test.ts:88]: Thank you for the updates @litewarp!
One last thing then this is good to go from my side, here and on line 124 please use `contextValue: neo4j.getContextValues()` that the query is executed against the correct database. Thanks!

**Review by angrykoala — APPROVED** [review]
Looks good to me. Thanks for this fix @litewarp

**Pull request merged**